govcd/vdc_template.go

```diff
@@ -0,0 +1,219 @@
+		}
+		if len(results.Results.OrgVdcTemplateRecord) > 1 {
+			return nil, fmt.Errorf("expected one VDC Template with name '%s', but got %d", name, len(results.Results.OrgVdcTemplateRecord))
+		}
+		return vcdClient.GetVdcTemplateById(results.Results.OrgVdcTemplateRecord[0].HREF)
+	}
+
+}
+
+// Delete deletes the receiver VDC Template
+func (vdcTemplate *VdcTemplate) Delete() error {
+	if !vdcTemplate.client.IsSysAdmin {
+		return fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return fmt.Errorf("cannot delete the VDC Template, its HREF is empty")
+	}
+
+	_, err := vdcTemplate.client.ExecuteRequest(vdcTemplate.VdcTemplate.HREF, http.MethodDelete, "", "error deleting VDC Template: %s", nil, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetAccess sets the Access control configuration for the receiver VDC Template,
+// which specifies which Organizations can read it.
+func (vdcTemplate *VdcTemplate) SetAccess(organizationIds []string) error {
+	if !vdcTemplate.client.IsSysAdmin {
+		return fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return fmt.Errorf("cannot set the Access list for the VDC Template, its HREF is empty")
+	}
+	accessSettings := make([]*types.AccessSetting, len(organizationIds))
+	for i, organizationId := range organizationIds {
+		accessSettings[i] = &types.AccessSetting{
+			Subject: &types.LocalSubject{
+				HREF: fmt.Sprintf("%s/org/%s", vdcTemplate.client.VCDHREF.String(), extractUuid(organizationId))},
+			AccessLevel: types.ControlAccessReadOnly,
+		}
+	}
+	payload := &types.ControlAccessParams{AccessSettings: &types.AccessSettingList{AccessSetting: accessSettings}}
+
+	return vdcTemplate.client.setAccessControlWithHttpMethod(http.MethodPut, payload, vdcTemplate.VdcTemplate.HREF, "VDC Template", vdcTemplate.VdcTemplate.Name, nil)
+}
+
+// GetAccess retrieves the Control access configuration for the receiver VDC Template, which
+// contains the Organizations that can read it.
+func (vdcTemplate *VdcTemplate) GetAccess() (*types.ControlAccessParams, error) {
+	if !vdcTemplate.client.IsSysAdmin {
+		return nil, fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return nil, fmt.Errorf("cannot get the Access list for the VDC Template, its HREF is empty")
+	}
+	result := &types.ControlAccessParams{}
+	href := fmt.Sprintf("%s/controlAccess", vdcTemplate.VdcTemplate.HREF)
+	_, err := vdcTemplate.client.ExecuteRequest(href, http.MethodGet, types.AnyXMLMime, "error getting access of VDC Template: %s", nil, result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// Instantiate creates a new VDC from the template and returns its ID if the operation finishes successfully.
+func (vdcTemplate *VdcTemplate) Instantiate(vdcName, description, organizationId string) (string, error) {
```

Comment:
I have a few concerns about this method:
1. The name suggests that the template is doing something, but not exactly what. Only the comment reveals that it is creating a VDC. Why not calling it `CreateVdcFromTemplate`? Or, since it is a method of `VdcTemplate`, perhaps `CreateVdc`?
2. the method returns a string and error. This is unusual for our creation methods. Either return a task+error ( like many "Async*" creation methods do) or a pointer to the created object. If there are problems with this strategy, they should be explained in a comment.